safe-global/safe-wallet-web
# [Counterfactual] Add backup option

## What it solves

Resolves #3203

## How this PR fixes it

- Shows a backup option when creating a counterfactual safe
- Falls back to read-only provider to fetch balance in case no wallet is connected and caches the response

## How to test it

1. Create a counterfactual safe
2. Observe the creation modal with a new section to download a backup
3. Download the backup
4. Disconnect your wallet
5. Reload the app
6. Observe balances still load once
7. Observe no polling for balances

## Screenshots

<img width="645" alt="Screenshot 2024-02-05 at 12 25 23" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/149a6ff2-6a96-4edb-95c0-3fa00da4ab20">


## Checklist
